src/app/auth/register/register.component.ts

```diff
@@ -1,16 +1,74 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, ViewChild } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { CustomValidators } from '../services/customValidators';
+import { UserService } from '../services/user.service';
+import { AuthService } from '../services/auth.service';
+import { ModalDirective } from 'ngx-bootstrap/modal';
 
 @Component({
-  selector: 'app-register',
   templateUrl: './register.component.html',
   styleUrls: ['./register.component.scss'],
 })
-export class RegisterComponent implements OnInit {
+export class RegisterComponent {
+  public frmSignup: FormGroup;
+  public massage: string;
+  @ViewChild('modal', {static: false}) modal: ModalDirective;
 
-  constructor() {
+  constructor(public userService: UserService, private fb: FormBuilder, private authService: AuthService) {
```

Comment:
always use full names for the service to make it clear, so formBuilder not fb
and when you have to inject more than two, make each in it's own line
```suggestion
  constructor(
    public userService: UserService,
    private formBuilder: FormBuilder,
    private authService: AuthService,
  ) {
```